streams/src/test/java/org/apache/kafka/streams/integration/LagFetchIntegrationTest.java

```diff
@@ -277,6 +277,7 @@ public void shouldFetchLagsDuringRestoration() throws Exception {
         t1.toStream().to(outputTopicName);
         final KafkaStreams streams = new KafkaStreams(builder.build(), props);
```

Comment:
We need to `close()` a `KafkaStreams` client even if we never `.start()` it, and thus should use `try-with-resource` or split this up and do:
```
final KafkaStreams streams;
try {
  streams = new KafkaStreams(...);
} finally {
  ...
}
```

Reply:
Actually the Streams clients are closed in the `finally`-clause. The thing that was missing was that the Streams clients were not created in the `try`-clause. I needed to use `AtomicReference` for that, otherwise I could not use the Streams clients in lambdas.